web/tests/rangelist.spec.js

```diff
@@ -1,26 +1,6 @@
 import RangeList from '../src/utils/rangelist';
 
 describe('RangeList', () => {
-  it('Produces non-overlapping ranges', () => {
```

Comment:
why did you delete all those tests?

Reply:
Because of the implementation change, this test case no longer applies.  Overlap is not meaningful for a hash-based approach:

Before, `this.members = [ 1, 2, 2, 3, 4, 5 ]` was a possible edge case this test guarded against.

Reply:
imo this is still a valid test. I don't get why it shouldn't pass anymore. And the case you are describing is still part of the specification of the RangeList class.